docs/reference/server-events.md

```diff
@@ -673,6 +673,29 @@ There is no data included with the event for handlers.
 
 Events emitted by the `@apostrophecms/page` module.
 
+### `beforeSend`
+
+Triggered just before a page template is rendered and sent as a response to a request. This can be a good place to adjust the data available to the template  by amending `req.data`.
```

Comment:
See also async components, which can be a cleaner way to package code that fetches additional data during page rendering.

Reply:
(I'm not suggesting we should deprecate or not document beforeSend. It still has its place I think. Just a little hard to nail down exactly when that is when async components are also available. We'll let best practices evolve)